### DISCUSSION TITLE
Change the PowerShell Version in PowerShell's Package Manager

### DISCUSSION THREAD

**Discussion by thiagojramos** [Devolutions/UniGetUI#2419] [General]
Is it possible to set a different PowerShell version in the "PowerShell's Package Manager"?

Currently, all the PowerShell executables I use (present in the PATH) are from `PowerShell 7.5.0-preview.3`. However, the "PowerShell's Package Manager" is using Windows PowerShell 5.1, which is not even defined in my PATH or as a system variable.

<details><summary>Screenshot: default PowerShell</summary>
<p>

![2024-07-03 (25)](https://github.com/marticliment/UniGetUI/assets/45890502/26c77e34-95a7-4818-9e7c-2be2332244c5)

</p>
</details> 

In the "PowerShell's Package Manager," the path to the Windows PowerShell executable appears, but there is no option to edit it, only to copy the path.

When adding a new source in "Manage PowerShell sources" (I tried to add PSGallery), the operation failed. For me, the "PowerShell's Package Manager" would only be useful if it could be used with PowerShell 7.

<details><summary>Screenshot: Failed to add PS Gallery</summary>
<p>

![2024-07-03 (26)](https://github.com/marticliment/UniGetUI/assets/45890502/f396bc81-4437-4e2e-98f0-6b88bc7f36ae)

</p>
</details> 

Here is the error log that appears when trying to add PSGallery as a source:

<details><summary>Error Log</summary>
<p>

```powershell
Starting source addition operation for source name=PSGallery with Manager name=PowerShell
Process Executable   : C:\WINDOWS\system32\windowspowershell\v1.0\powershell.exe
Process Call Arguments : -NoProfile -Command Register-PSRepository -Default
Working Directory   : C:\Users\thiag
Process Start Time   : 07/03/2024 10:37:19 PM
ERR | Register-PSRepository : The 'Register-PSRepository' command was found in the 'PowerShellGet' module, but the module could not be loaded. For more information, run 'Import-Module PowerShellGet'.
ERR | At line:1 char:1
ERR | + Register-PSRepository -Default
ERR | + ~~~~~~~~~~~~~~~~~~~~~
ERR |   + CategoryInfo     : ObjectNotFound: (Register-PSRepository:String) [], CommandNotFoundException
ERR |   + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule
Process Exit Code   : 1
Process End Time    : 07/03/2024 10:37:20 PM
``` 

</p>
</details> 

UniGetUI Version: 3.1.0 stable (the latest release)

Any suggestions or solutions?

Note: I was going to create a "New issue," but I wasn't sure which category my case would fit into. So, I'll post it here.

**Comment by marticliment** [comment]
There is an issue for this here: https://github.com/marticliment/UniGetUI/issues/1994

**Reply by thiagojramos** [reply]
thanks
